src/Zicht/Bundle/UrlBundle/Url/AliasSitemapProvider.php

```diff
@@ -35,35 +36,24 @@ public function __construct(Connection $connection, EventDispatcherInterface $ev
     /**
      * @{inheritDoc}
      */
-    public function all(AuthorizationCheckerInterface $authorizationChecker): array
+    public function all(AuthorizationCheckerInterface $authorizationChecker)
     {
         $query = $this->connection->prepare('SELECT * FROM url_alias WHERE mode=?');
         $query->execute([UrlAlias::REWRITE]);
-
-        $urls = iterable($query->fetchAll(\PDO::FETCH_ASSOC));
+        $urls = new \ArrayObject($query->fetchAll(\PDO::FETCH_ASSOC));
 
         /**
          * Hook to allow the mapping to be modified at run-time.
          */
         if ($this->eventDispatcher->hasListeners(Events::EVENT_SITEMAP_FILTER)) {
-            $event = $this->eventDispatcher->dispatch(
-                Events::EVENT_SITEMAP_FILTER,
-                new SitemapFilterEvent(
-                    $urls, [
-                        'authorization_checker' => $authorizationChecker,
-                        'connection' => $this->connection,
-                    ]
-                )
-            );
-
-            $urls = $event->getSubject();
+            $this->eventDispatcher->dispatch(Events::EVENT_SITEMAP_FILTER, new SitemapFilterEvent($urls));
```

Comment:
This changes the API as it is now used in one of our projects, so this is either a breaking change (requiring a new major version) or the project needs to be changes accordingly.  I think the best solution, by far, is changing the project.

Reply:
agree, and project version should be locked so a heads up should be fine?